src/Item.js

```diff
@@ -22,12 +24,29 @@ function Item(props) {
         'formData.purchaseDates': purchaseDates,
       });
     } else {
-      const date = new Date();
-      db.collection(userToken)
-        .doc(id)
-        .update({
-          'formData.purchaseDates': [...purchaseDates, date],
-        });
+      const newDate = new Date();
+      const newDates = [...purchaseDates, newDate];
+      const numberOfPurchases = newDates.length;
+      let latestInterval = null;
+
+      if (numberOfPurchases >= 2) {
+        const lastDate = fromUnixTime(
+          purchaseDates[purchaseDates.length - 1].seconds,
+        );
+        latestInterval = intervalToDuration({
+          start: lastDate,
+          end: newDate,
+        }).days;
+      }
+      const newInterval = estimates(
+        nextEstimate,
+        latestInterval,
+        numberOfPurchases,
+      );
+      db.collection(userToken).doc(id).update({
+        'formData.purchaseDates': newDates,
+        'formData.nextEstimate': newInterval,
+      });
     }
     setChecked(!checked);
   };
```

Comment:
Great work with this logic, you two! It made my head spin trying to think about how to do this - looking forward to hearing more about it on Saturday!